website/content/docs/api-gateway/configuration/gateway.mdx

```diff
@@ -0,0 +1,316 @@
+* matchLabels
+
+### listeners.allowedRoutes.namespaces.selector.matchExpressions
+Specifies an array of requirements for matching namespaces. If a match is found, then routes from the matching namespace(s) are allowed to attach to the `Gateway`. The following table describes members of the `matchExpressions` array:
+
+| Requirement |                                                                                                                       Description                                                                                                                      |       Type       |
+|:-----------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:----------------:|
+| key         | Specifies that label that the key applies to.                                                                                                                                                                                                          | string           |
+| operator    | Specifies the key's relation to a set of values. The following values are valid:In: description of what this means NotIn: description of what this means Exists: description of what this means DoesNotExist: description of what this means           | string           |
+| values      | Specifies an array of string values. If the operator is configured to In or NotIn,the values array must be non-empty. If theoperator is Exists or DoesNotExist, the values array must be empty. This array is replaced during a strategic merge patch. | array of strings |
+| scope       | Specifies the effective scope of the Gateway. The value should always be namespaced.                                                                                                                                                                   | Required         |
+| fields      | Specifies the configurations for the Gateway. The fields are listed in the Configuration model. Details for each field are described in the Specification.                                                                                             | Required         |
+
+```yaml
+namespaceSelector:
+  matchExpressions:
+  - key: kubernetes.io/metadata.name
+    operator: In
+    values:
+    - foo
+    - bar
+```
+
+### listeners.allowedRoutes.namespaces.selector.matchLabels
+Specifies an array of labels and label values. If a match is found, then routes with the matching label(s) are allowed to attach to the `Gateway`. This selector can contain any arbitrary key/value pair.
+
+```yaml
+namespaceSelector:
+    matchLabels:
+      foo: bar
+```
+
+For more on labels, see [Labels and Selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
+
+### listeners.hostname
+Specifies the `listeners` hostname
```

Comment:
```suggestion
Specifies the `listener`'s hostname
```

Reply:
Is there any information we could provide about what the `hostname` is used for?